src/wasm-type.h

```diff
@@ -202,18 +203,28 @@ class Type {
   // Returns true if left is a subtype of right. Subtype includes itself.
   static bool isSubType(Type left, Type right);
 
-  // Computes the least upper bound from the type lattice.
-  // If one of the type is unreachable, the other type becomes the result. If
-  // the common supertype does not exist, returns none, a poison value.
-  static Type getLeastUpperBound(Type a, Type b);
-
-  // Computes the least upper bound for all types in the given list.
-  template<typename T> static Type mergeTypes(const T& types) {
-    Type type = Type::unreachable;
-    for (auto other : types) {
-      type = Type::getLeastUpperBound(type, other);
+  // Returns true and sets `super` to the type in range [begin,end) that is a
+  // supertype of all types in the range iff such a type exists. This is similar
+  // to but more limited than calculating a proper least upper bound on the
+  // types.
+  // FIXME: This is not necessarily deterministic as long as we are not
+  // canonicalizing structurally identical recursive types because it may end up
+  // choosing any of multiple identical candidates.
```

Comment:
could we do a stable sort here to make this deterministic? then the result would be determined by the order of the inputs, which is enough.

Reply:
Yes, I think that if we use a `stable_sort` here and use `std::set` instead of `std::unordered_set` to collect types before calling this function, that would be sufficient to make it deterministic. I'll do that.